packages/mui-base/src/NumberField/utils/parse.ts

```diff
@@ -0,0 +1,39 @@
+export const HAN_NUMERALS = ['零', '一', '二', '三', '四', '五', '六', '七', '八', '九'];
+export const ARABIC_NUMERALS = ['٠', '١', '٢', '٣', '٤', '٥', '٦', '٧', '٨', '٩'];
+
+export function getNumberLocaleDetails(options?: Intl.NumberFormatOptions) {
+  const parts = new Intl.NumberFormat(undefined, options).formatToParts(1111.1);
```

Comment:
I don't see this working at scale, the locale is set to `undefined`, so per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#locales, it uses the "default locale". So

- Developers can't force a locale, e.g. fr-FR
- During SSR hydration, there could be a mismatch, either because the server doesn't have the same CLDR data as the browser or because the locale doesn't match (default locale of the browser is fr-FR, default locale of the server is en-US)

Reply:
I wasn't sure if they should be able to force a locale. It seems like an accessibility issue in that it should probably always display how the user's preference wants it to?

Reply:
One case where it might be useful is during SSR via the [Accept-Language](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language) header. They could set a `locale` prop on the component using this, e.g. in a React Server Component?